src/dmd/opover.d

```diff
@@ -213,9 +213,12 @@ private Expression checkAliasThisForLhs(AggregateDeclaration ad, Scope* sc, BinE
     if (isRecursiveAliasThis(e.att1, e.e1.type))
         return null;
     //printf("att %s e1 = %s\n", Token::toChars(e.op), e.e1.type.toChars());
-    Expression e1 = new DotIdExp(e.loc, e.e1, ad.aliasthis.ident);
     BinExp be = cast(BinExp)e.copy();
-    be.e1 = e1;
+    // Resolve 'alias this' but in case of assigment don't resolve properties yet
+    // because 'e1 = e2' could mean 'e1(e2)' or 'e1() = e2'
+    be.e1 = resolveAliasThis(sc, e.e1, true, e.op == TOK.assign);
```

Comment:
you could assign a variable `bool noAliasThis = (e.op == TOK.assign)`
that would make it clearer what's going on.

Reply:
See I already got it wrong.
I meant `bool findOnly = (e.op == TOK.assign)`

Reply:
good idea